khmer/_oxli/graphs.pyx

```diff
@@ -211,87 +212,78 @@ cdef class Hashtable:
                                                                    max_count))
         return posns
 
-    def consume_seqfile_with_reads_parser(self, read_parser):
-        """Count all k-mers from read_parser."""
-        cdef unsigned long long n_consumed = 0
-        cdef unsigned int total_reads = 0
-
-        cdef CPyReadParser_Object* parser = <CPyReadParser_Object*>read_parser
+    cdef FastxParserPtr _get_parser(self, object parser_or_filename) except *:
+        cdef FastxParserPtr _parser
+        if type(parser_or_filename) is FastxParser:
+            _parser = (<FastxParser>parser_or_filename)._this
+        elif type(parser_or_filename) is ReadParser:
+            _parser = #  Unholy incantation
```

Comment:
I tried several things but I can't figure out the right incantation to go here. If you have a few minutes tomorrow to pair program this tomorrow @camillescott that would be wonderful. Unless it's a trivial fix, in which case a comment here should be fine.

Reply:
`_parser = (<CPyReadParser_Object>parser_or_filename).parser` ought to do it, along with a `from khmer._oxli.parsing cimport CPyReadParser_Object` up top

Reply:
Thanks!

Now clang is choking on the Cythonized C++ code. It looks like cython is using C++17 features. 😒

```
...
...
...
khmer/_oxli/graphs.cpp:16623:9: warning: use of the 'fallthrough' attribute is a C++1z extension [-Wc++1z-extensions]
        CYTHON_FALLTHROUGH;
        ^
khmer/_oxli/graphs.cpp:481:36: note: expanded from macro 'CYTHON_FALLTHROUGH'
      #define CYTHON_FALLTHROUGH [[fallthrough]]
                                   ^
khmer/_oxli/graphs.cpp:16636:9: warning: use of the 'fallthrough' attribute is a C++1z extension [-Wc++1z-extensions]
        CYTHON_FALLTHROUGH;
        ^
khmer/_oxli/graphs.cpp:481:36: note: expanded from macro 'CYTHON_FALLTHROUGH'
      #define CYTHON_FALLTHROUGH [[fallthrough]]
                                   ^
168 warnings and 2 errors generated.
error: command 'clang' failed with exit status 1
make: *** [test] Error 1
```

Reply:
Never mind, I didn't see the trees for the forest. These warnings swamped the real issues.

```
...
...
...
khmer/_oxli/graphs.cpp:5723:43: error: member reference type 'PyObject *' (aka '_object *') is a pointer; did you mean to use '->'?
    __pyx_t_5 = __pyx_v_parser_or_filename.parser;                                                                                            
                ~~~~~~~~~~~~~~~~~~~~~~~~~~^                                                                          
                                          ->
khmer/_oxli/graphs.cpp:5723:44: error: no member named 'parser' in '_object'
    __pyx_t_5 = __pyx_v_parser_or_filename.parser;
                ~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
...
...
...
```

This was my experience last night. I couldn't figure out how to handle objects vs pointers etc.

Reply:
Just needed to add a \* to `_parser = (<CPyReadParser_Object>parser_or_filename).parser` to make it `_parser = (<CPyReadParser_Object*>parser_or_filename).parser`. w00t!